doc/devices/dts/device_tree.rst

```diff
@@ -80,6 +80,59 @@ This .fixup file by default resides in the board directory and is named
 dts.fixup.  This fixup file maps the generated include information to the
 current driver/source usage.
 
+.. _dt_vs_kconfig:
+
+Device Tree vs Kconfig
+**********************
+
+As mentioned above there are several tools used to configure the build with
+Zephyr.
+The two main ones, Device Tree and Kconfig, might seem to overlap in purpose,
+but in fact they do not. This section inteds to serve as a reference when
+deciding whether to place a certain configuration item in Device Tree or Kconfig.
+
+The scope of each configuration tool can be summarized as follows:
+
+* Device Tree is used to describe the **hardware** and its **boot-time
+  configuration**.
+* Kconfig is used to describe which **software features** will be built into
+  the final image, and their **configuration**.
+
+Hence Device Tree deals mostly with hardware and Kconfig with software. A
+noteworthy exception is Device Tree's `chosen` keyword, which allows the user
+to select a particular instance of a hardware device to be used for a concrete
+purpose by the software. An example of this is using this keyword to select
+a particular UART to be used as the system's console.
+
+In order to further clarify this separation, let's use a particular, well-known
+example to illustrate this: serial ports a.k.a. UARTs. Let's consider a
+board containing a SoC with 2 UART instances:
+
+* The fact that the target hardware **contains** 2 UARTs is described with Device
+  Tree. This includes the UART type, its driver compatibility and certain
+  "immutable" (i.e. not software-configurable) settings such as the base address
+  of the hardware peripheral in memory or its interrupt line.
+* Additional, **hardware boot-time configuration** is also described with Device
+  Tree. This includes things like the interrupt line and IRQ priority. These
+  may also be modifiable at runtime later, but their boot-time "default"
+  configuration is described in Device Tree.
+* The fact that the user intends to include **software support** for UART in the
+  build is described in Kconfig. Through the use of Kconfig, users can opt not
+  to include support for this particular hardware item if they don't require it.
```

Comment:
Maybe this part should be clarified as it eludes the device tree status property.
Should we use it?
If we use it (which I think we should do), we should mention it will take precedence and then Kconfig will be able to amend only if status is "okay". If status is "disabled" it won't be available in Kconfig.
If we don't use it, we'll need to change the (device tree extract scripts) code.